server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -379,6 +379,7 @@ public ClusterState execute(ClusterState currentState) {
                     if (request.includeGlobalState()) {
                         if (metaData.persistentSettings() != null) {
                             Settings settings = metaData.persistentSettings();
+                            clusterSettings.validate(settings, true);
```

Comment:
I'm afraid I might know too little about settings updates here, but couldn't we be in a situation where the settings as found in the repo will not validate, but they will validate after updates have been executed on them?

Reply:
@original-brownbear I'm not sure either. @ywelsch what do you think?